api/src/org/apache/cloudstack/api/command/admin/host/ListHostsCmd.java

```diff
@@ -203,9 +201,22 @@ public void execute() {
                 hostResponse.setObjectName("host");
                 hostResponses.add(hostResponse);
             }
-
             response.setResponses(hostResponses, result.second());
         }
+        // Remove sensitive details from host response
+        List<HostResponse> hostResponsesList = response.getResponses();
+        for (HostResponse hostResponse: hostResponsesList) {
+            Map<String, String> hostDetails = hostResponse.getDetails();
+            if (hostDetails == null) continue;
+            if (hostDetails.containsKey("username")) {
+                hostDetails.remove("username");
+            }
+            if (hostDetails.containsKey("password")) {
+                hostDetails.remove("password");
+            }
+            hostResponse.setDetails(hostDetails);
```

Comment:
@jburwell The HostResponse created by service layer is being consumed at several places such as AddHostCmd, UpdateHostCmd, ReconnectHostCmd etc. while the details are set in HostJoinDaoImpl (much lower layer than the service layer), this is why I thought the best fix would be to fix the response in ListHostsCmd before it is wired back to the client.

The response class gets instantiated in the current request thread and is not referenced by any other threads or components, that's why it's safe to simply manipulate the map in the response class without adding the overhead of copying it. It's safe in this context, though I agree with the immutable pattern you're proposing here - I can add code to copy the details map instead of manipulating it.